src/frontend/src/components/public/PIANavButton/index.tsx

```diff
@@ -49,18 +52,19 @@ const PIANavButton = ({ pages }: INavButton) => {
               : 'Back'}
           </button>
         )}
-        {HandleState(pages, pathname, 'next') && (
-          <button
-            className="bcgovbtn bcgovbtn__secondary btn-next ms-auto"
-            onClick={handleNext}
-            type="button"
-            aria-label="Next Button"
-          >
-            {typeof HandleState(pages, pathname, 'next') === 'object'
-              ? Object(HandleState(pages, pathname, 'next'))?.title
-              : 'Next'}
-          </button>
-        )}
+        {HandleState(pages, pathname, 'next') &&
+          (!isDelegate || userRole !== undefined) && (
```

Comment:
should not this whole logic be processed inside handlestate?

Reply:
The HandleState wasn't handling this case at all and we weren't even using the isDelegate prop which we were already passing in. I see the point you are making but I really don't think I should mess with the HandleState as it doesn't currently support these checks and I don't want to have to spend too much time testing for side effects / debugging. Happy to discuss in a call tomorrow (I'm totally booked today) if this is still an issue.

Reply:
Makes sense. Approved!